src/vue/forms/LoginForm.spec.js

```diff
@@ -0,0 +1,188 @@
+import { TestHelper } from '@/test/test-helper'
+import LoginForm from './LoginForm'
+
+import Vuelidate from 'vuelidate'
+import Vuex from 'vuex'
+import Vue from 'vue'
+
+import { Wallet, errors } from '@tokend/js-sdk'
+import { vuexTypes } from '@/vuex'
+import { createLocalVue, shallowMount, mount } from '@vue/test-utils'
+import { globalize } from '@/vue/filters/globalize'
+
+import { MockHelper } from '@/test'
+
+// HACK: https://github.com/vuejs/vue-test-utils/issues/532, waiting for
+// Vue 2.6 so everything get fixed
+Vue.config.silent = true
+
+const localVue = createLocalVue()
+localVue.use(Vuelidate)
+localVue.use(Vuex)
+localVue.filter('globalize', globalize)
+
+describe('LoginForm component unit test', () => {
+  let mockHelper
+  beforeEach(() => {
+    sinon.restore()
+    mockHelper = new MockHelper()
+  })
+
+  describe('validation works correctly', () => {
+    let wrapper
+
+    beforeEach(() => {
+      wrapper = mount(LoginForm, { localVue })
+    })
+
+    const fields = {
+      email: {
+        valid: ['anything', 'alice@mail.com', 'qq'],
+        invalid: ['']
+      },
+      password: {
+        valid: ['anything', 'q', 'qqww'],
+        invalid: ['']
+      }
+    }
+
+    for (const [fieldName, fieldValues] of Object.entries(fields)) {
+      for (const fieldValue of fieldValues.valid) {
+        it(`considers ${fieldValue} a valid ${fieldName}`, () => {
+          expect(TestHelper.isFieldValid(
+            wrapper,
+            fieldName,
+            fieldValue
+          ))
+            .to.be.true
+        })
+      }
+      for (const fieldValue of fieldValues.invalid) {
+        it(`considers ${fieldValue} an invalid ${fieldName}`, () => {
+          expect(TestHelper.isFieldValid(
+            wrapper,
+            fieldName,
+            fieldValue
+          ))
+            .to.be.false
+        })
+      }
+    }
+
+    const fieldBindings = {
+      '#login-email': 'form.email',
+      '#login-password': 'form.password'
+    }
+
+    for (const [selector, model] of Object.entries(fieldBindings)) {
+      it(`$v.${model} is touched after blur event emitted on ${selector}`, async () => {
+        const touchField = sinon.spy()
+
+        wrapper.setMethods({ touchField })
+        wrapper.find(selector).vm.$emit('blur')
+
+        expect(touchField.withArgs(model).calledOnce).to.be.true
+      })
+    }
+  })
+
+  describe('methods', () => {
+    let spies
+    let wrapper
+    let sdk = {
+      api: {
+        users: {}
+      }
+    }
+
+    beforeEach(() => {
+      spies = {
+        loadWallet: sinon.spy()
+      }
+
+      const store = new Vuex.Store({
+        modules: {
+          'new-wallet': {
+            namespaced: true,
+            actions: {
+              [vuexTypes.LOAD_WALLET]: spies.loadWallet
+            },
+            getters: {
+              [vuexTypes.wallet]: sinon.stub().returns(new Wallet(
```

Comment:
you got `mockWallet` in the `mockHelper`. isn’t it unused anymore?

Reply:
resolved, updated